src/Core/Simplify.hs

```diff
@@ -626,19 +626,18 @@ effectLabelFromHandler tp
   = -- fromHandlerName
     (labelName tp)
 
-effectOffset :: Name -> Type -> Integer
+effectOffset :: (Int, Name) -> Type -> Integer
 effectOffset l effTp
   = let (ls,_) = extractHandledEffect effTp
         ofs = findMatch 0 l ls
     in -- trace ("found offset " ++ show ofs ++ " for " ++ show l ++ " in " ++ show (map (show . labelName) ls)) $
        ofs
 
-findMatch :: Integer -> Name -> [Type] -> Integer
-findMatch i lname (l:ls)  = if (labelName l == lname) then i else findMatch (i+1) lname ls
+findMatch :: Integer -> (Int, Name) -> [(Int, Type)] -> Integer
+findMatch i (off, lname) ((ol,l):ls)  = if (labelName l == lname) && off == ol then i else findMatch (i+1) (off, lname) ls
```

Comment:
This changes to find the label with the correct name at the correct count of the label name.